lib/SILOptimizer/Mandatory/DefiniteInitialization.cpp

```diff
@@ -810,15 +812,54 @@ static void injectHopToExecutorAfter(SILLocation loc,
   // initialized.
   assert(!isa<TermInst>(*insertPt) && "unexpected hop-inject after terminator");
 
-  SILBuilderWithScope::insertAfter(&*insertPt, [&](SILBuilder &b) {
-    if (needsBorrow)
-      actor = b.createBeginBorrow(loc, actor);
+  auto injectAfter = [&](SILInstruction *insertPt) -> void {
+    LLVM_DEBUG(llvm::dbgs() << "hop-injector: injecting after " << *insertPt);
+    SILBuilderWithScope::insertAfter(insertPt, [&](SILBuilder &b) {
+      if (needsBorrow)
+        actor = b.createBeginBorrow(loc, actor);
+
+      b.createHopToExecutor(loc, actor, /*mandatory=*/false);
+
+      if (needsBorrow)
+        b.createEndBorrow(loc, actor);
+    });
+  };
+
+  //////
+  // FIXME: It is invalid for a hop to appear within an access scope, since the
```

Comment:
I'm not sure what this FIXME is for. Should we say:
This is a best-effort to avoid placing hops within an access, because we would prefer the access set to be empty.
But the hop may still be emitted in an overlapping access in theory, and the runtime can handle that?

It would be awesome to come up with a test case for overlapping access, but I don't know how to do it given that properties can neither be initialized using 'inout' nor with a closure

Reply:
Okay, I'll change it to a NOTE explaining the rationale for doing this extra work.